test/flow_connector_path/down_forward_down_backward_up_path/component_test.js

```diff
@@ -0,0 +1,94 @@
+require("../../setup");
+
+describe("DownForwardDownBackwardUpPath", function() {
+  const helpers = require("../helpers.js");
+  const c = helpers.constants;
+  const CONTAINER_ID = "downforwarddownbackwarduppath-for-testing-component-container";
+  const COMPONENT_ID = "downforwarddownbackwarduppath-for-testing-component-connector";
+
+  describe("Component", function() {
+    var created;
+    const POINTS = {
+                     from_x: 1951,
+                     from_y: 125,
+                     to_x: 1850,
+                     to_y: 562.5,
+                     via_x: 525,
+                     via_y: 312.5
```

Comment:
I'm very willing for this to be me no understanding how 'via' works, but I don;t see how these co-ordinates result in a `DownForwardDownBackwardUpPath`

Reply:
I understand why you would think this but I believe it is the result of needing to calculate element position/dimensions in order to get the via_x/y coordinate. Essentially, it is relative to the starting (from_x/y) point, not the container.

**Longer explanation**
Since the from_x/y and and to_x/y show it needs to go backward, but the design says it needs to go forward first (it's a branching line so needs to pass through end of condition comment element), the via_x/y is a point that tells it how far to go forward (somewhere around the end of that comment element), before it goes backward, if you see what I mean.

It's the via_x/y is the point between forward and down, in the path, and is relative to the from_x/y, not the whole container (which I think from_x/y and to_x/y are - so that's why they are larger numbers). It could probably be calculated using it's position from the container, but that is a larger calculation than would also involve subtracting the starting (from_x/y) point, just to see how far forward the line should go. Instead it's pretty much the value we want (but we need also to subtract the curve size numbers).

I tapped into the code to output the points data of a real paths to get (not calculate - I'm not that smart!) most of these testing points you see in the code. I've just done that again for this DownForwardDownBackwardUpPath example to show them in the console (similar to what you see above). I've also attached a couple screenshots that hopefully help to illustrate what I'm trying to explain in case none of this is very clear.

**Path showing points data and element outline**
<img width="1766" alt="Screenshot 2022-08-31 at 11 36 20" src="https://user-images.githubusercontent.com/76942244/187662510-d1c5e886-08e6-4256-9761-9cfef91156f8.png">

**Browser ruler showing (approximate) dimensions of the via_x/y position** (bottom right blue area)
<img width="982" alt="Screenshot 2022-08-31 at 11 36 50" src="https://user-images.githubusercontent.com/76942244/187662614-ed02b574-cb40-4863-b462-57f67baa6796.png">